<launch>

    <!-- Parameters to define true pose for reference in error calculations-->
    <param name="true_x" type="double" value="0"/>
    <param name="true_y" type="double" value="0"/>
    <param name="true_z" type="double" value="1"/>
    <param name="true_orientation_x" type="double" value="0"/>
    <param name="true_orientation_y" type="double" value="0"/>
    <param name="true_orientation_z" type="double" value="0"/>
    <param name="true_orientation_w" type="double" value="1"/>
    <param name="total_data_points" type="int" value="500"/>

    <!-- parameters for image_publisher node -->
    <param name="device_id" value="0"/>
    <param name="calibration_file" value="camera_a.yaml"/>
    <param name="publish_rate" value="20"/>
    <param name="image_width" value="640"/>
    <param name="image_height" value="480"/>

    <!-- Run all the nodes required for the ekf test -->
    <node name="image_publisher" pkg="robot_sensors" type="image_publisher.py">
        <param name="board_x_markers" type="int" value="5"/>
        <param name="board_y_markers" type="int" value="7"/>
        <param name="board_marker_length" type="double" value=".029"/>
        <param name="board_marker_separation" type="double" value=".006"/>
    </node>
    <node name="aruco" pkg="robot_slam" type="aruco_localization.py"/>
    <node name="aruco_ekf" pkg="robot_slam" type="ArucoEKF.py"/>
    <node name="test_ekf" pkg="robot_slam" type="testEKF.py"/>

    <!-- Launch the rqt_console to view the log info printed at the end of the test -->
    <node name="rqt_console" pkg="rqt_console" type="rqt_console"/>

</launch>
